src/utils/matching/matchRequestUrl.ts

```diff
@@ -42,6 +42,11 @@ export function coercePath(path: string): string {
        * @see https://github.com/pillarjs/path-to-regexp/issues/259
        */
       .replace(/^([^\/]+)(:)(?=\/\/)/g, '$1\\$2')
+      /**
+       * Escape the port so that "path-to-regexp" can match
+       * absolute URLs including port numbers.
+       */
+      .replace(/((?::)(?:[0-9]+))/g, '\\$1')
```

Comment:
Alternatively the following should also work `/(:[0-9]+)/`. @kettanaito you can pick :)